picard/match_details_log.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+# Copyright (C) 2020 Ray Bouchard
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+import numbers
+import traceback
+
+from picard import log
+
+
+__all_file_result_selected_entry_nbr = {}  # dictionary of song file name and selected_entry
+__all_file_match_dtls = {}                 # dictionary of a song file and match detail entries associated with it.
+__cur_file = None
```

Comment:
I'm not sure to understand comments there.
What's a "song file"? What is the exact purpose of those dictionnaries?

I noticed you shorten variable and class names without good reason:
`dtls` ->  `details`
`lbl` -> `label`
`cur` -> `current`
`nbr` -> `number`
etc.

Reply:
"song file"  :) I couldn't think of a more descriptive name. its what you drag from left pane to the middle pane.  I could use just "file"?   These dictionaries are where we hold all the match data, and selections for all the files that have been "scanned" or "looked up"

Right, its common where I come from to use abbreviations.  Do you not do that?  Only guide I found was  Pep-0008 "use English words wherever feasible (in many cases, abbreviations and technical terms are used "  so seems its ok to use abbreviations.

Reply:
Ok, so "song file" is an audio file,  File in Picard (see https://github.com/metabrainz/picard/blob/3c3a4c2c8e39aef5c5b7030626a71ed317d6d1ea/picard/file.py#L88)

I see `__all_file_result_selected_entry_nbr[p_filename] = p_selected_entry_nbr` in the code, so keys for this dictionnary are file paths, right?
Since we already have those infos in `File` wouldn't make sense to store details in those objects?

> Right, its common where I come from to use abbreviations.

When it comes to Picard code, it is a bad practice, having an explicit name for a class, function or variable is (in most cases) better. Of course, they are exceptions, but using `dtls` instead `details` is hardly useful, and server absolutly no purpose nowadays (a long time ago length of variable names was limited by language, or available memory, but that's not the case anymore since a long long time).

Reply:
I updated the comment to reflect File instead of "song file".

I'll avoid the abbreviations going forward.

Yes we could track the selected_entry_nbr in the File class, as its at the same level of granularity.  but I think the solution is more modular this way.  I think its better to encapsulate the logic and state information for the match details in one module.  This variable holds state info for the match details rather than describing a File.  Without the Match Details Dialog, it would have no context/purpose.